### ISSUE TITLE
[Bug] You stumbled upon a bug in vite-plugin-ssr's source code

### ISSUE THREAD

**Issue by iMrDJAi** [brillout/vite-plugin-ssr#619]
### Description

I just updated to VPS v0.4.72 from v0.4.3, and I got that error as soon as I started my project.
Not sure what really happened. I'm posting the error stack as requested.

### Error Message + Error Stack

```shell
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: [vite-plugin-ssr@0.4.72][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Reach out at https://github.com/brillout/vite-plugin-ssr/issues/new or https://discord.com/invite/qTq92FQzKb and include this error stack (the error stack is usually enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust.
    at Proxy.renderPage (<project>\node_modules\vite-plugin-ssr\dist\cjs\node\runtime\renderPage.js:80:28)
    at async eval (eval at instantiateModule (file:///<project>/node_modules/vite/dist/node/chunks/dep-387affca.js:52388:28), <anonymous>:73:25)
```
```

**Comment by brillout** [comment]
Insert the following before the line `80` of `\node_modules\vite-plugin-ssr\dist\cjs\node\runtime\renderPage.js`.

```js
console.log('pageContextOriginal.httpResponse?.statusCode', pageContextOriginal.httpResponse?.statusCode);
console.log('statusCode', statusCode);
console.log('pageContextInit.urlOriginal', pageContextInit.urlOriginal);
```

What the output?

**Comment by iMrDJAi** [comment]
@brillout 
```
pageContextOriginal.httpResponse?.statusCode undefined
statusCode 200
pageContextInit.urlOriginal /
Waiting for the debugger to disconnect...
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^
```

**Comment by brillout** [comment]
Fix released in `0.4.73`.

I refactored the problematic piece of code; let me know if you still run into problems.

In case your company is up for it: https://github.com/sponsors/brillout &mdash; makes a big difference!

**Comment by iMrDJAi** [comment]
@brillout Thx! It's working perfectly now.
I'll definitely consider supporting this cool project sometime in the future!
